File: day01/day01.py
#!/usr/bin/env python3
import itertools
data = [int(x) for x in open("input.txt").readlines()]
print(sum(data))

freq = 0
seen = set([0])
for num in itertools.cycle(data):
    freq += num
    if freq in seen:
        print(freq); break
    seen.add(freq)
# print(next(f for f in accumulate(cycle(changes)) if f in seen or seen.add(f)))

File: day01/day01.rb
#!/usr/bin/env ruby
require 'set'
data = File.readlines('input.txt').map(&:to_i)
puts data.sum

freq = 0
seen = Set.new
data.cycle { |num|
  freq += num
  (puts freq; break) if not seen.add?(freq)
}
